src/utils/event-helper.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Whether addEventListener supports options or only takes passive as a boolean
+ * @type {boolean|undefined}
+ */
+export let passiveSupported;
+
+/**
+ * Return boolean. if listener option is supported, return `true`.
+ * if not supported, return `false`
+ * @param {!Window} win
+ * @return {boolean}
+ */
+export function supportsPassiveEventListener(win) {
+  if (passiveSupported !== undefined) {
+    return passiveSupported;
+  }
+
+  passiveSupported = false;
+  try {
+    const options = {
+      get passive() {
+        // This function will be called when the browser
+        // attempts to access the passive property.
+        passiveSupported = true;
+        return false;
+      },
+    };
+
+    win.addEventListener('test-options', null, options);
+    win.removeEventListener('test-options', null, options);
+  } catch (err) {
+    // EventListenerOptions are not supported
+  }
+  return passiveSupported;
+}
+
+/**
+ * Resets the test for whether addEventListener supports passive options or not.
+ */
+export function resetPassiveSupportedForTesting() {
+  passiveSupported = undefined;
+}
+
+supportsPassiveEventListener(self);
```

Comment:
Let's remove this function call to avoid immediate side effects when importing this module. Instead, use `supportsPassiveEventListener()`.